polyfill/intersection-observer.js

```diff
@@ -395,7 +399,10 @@ IntersectionObserver.prototype._monitorIntersections = function(doc) {
   });
 
   // Also monitor the parent.
-  if (doc != (this.root && this.root.ownerDocument || document)) {
+  if (
+    doc != ((this.root && this.root.ownerDocument) || document) &&
+    !isDoc(this.root)
```

Comment:
Is this a sufficient check? What if the `root` is a document, but in the parent subframe?

Reply:
thats interesting! I hadn't thought about how in same-origin nested iframes, the `root`  could be the document of a parent iframe. check should probably be closer to:

```js
var rootDoc = (this.root && (this.root.ownerDocument || this.root)) || document;
if (doc != rootDoc) {
   ...
} 

```

Reply:
fixed!